src/main/java/org/apache/hadoop/fs/PrestoFileSystemCache.java

```diff
@@ -77,6 +77,10 @@ FileSystem getUnique(URI uri, Configuration conf)
     private synchronized FileSystem getInternal(URI uri, Configuration conf, long unique)
             throws IOException
     {
+        synchronized (UserGroupInformation.class) {
+            UserGroupInformation.setConfiguration(conf);
```

Comment:
this is going the change the state of UGI. Are we sure that only UGI is for file system?
I think there should be an option to disable this.

Reply:
I may not totally get what you mean or understand the HDFS deeply. If so ,sorry about this.

The problem's root cause is that the `UserGroupInformation` class has a static variable: `private static Configuration conf`.This means secure hive and no secure hive share the same  'conf' value in `UserGroupInformation`. Based on this, the FileSystems are not  created rightly.Do you agree with this ?
 
Before we implement the classloader separation between catalogs,
I just want to make the state of `UserGroupInformation` is right before `FileSystems` are created. I tested my code , the result is fine.